features/mesh/dataplanes/DataplanePolicies.feature

```diff
@@ -1,61 +1,339 @@
+              toRules:
+                - matchers:
+                    - key: kuma.io/service
+                      not: false
+                      value: other-svc
+                  origin:
+                    - mesh: default
+                      name: the-other-http-route
+                      type: MeshHTTPRoute
+                - matchers:
+                    - key: kuma.io/service
+                      not: false
+                      value: backend_kuma-demo_svc_3001
+                  origin:
+                    - mesh: default
+                      name: the-http-route
+                      type: MeshHTTPRoute
+            - type: MeshTimeout
+              toRules:
+                - matchers:
+                    - key: __rule-matches-hash__
+                      not: false
+                      value: waFsoIISmZDTfFWYqxvY265/GASHYEWvHQTwMh/bpuU=
+                    - key: kuma.io/service
+                      not: false
+                      value: backend_kuma-demo_svc_3001
+                  origin:
+                    - mesh: default
+                      name: on-route
+                      type: MeshTimeout
+                    - mesh: default
+                      name: on-service
+                      type: MeshTimeout
+                - matchers:
+                    - key: __rule-matches-hash__
+                      not: true
+                      value: waFsoIISmZDTfFWYqxvY265/GASHYEWvHQTwMh/bpuU=
+                    - key: kuma.io/service
+                      not: false
+                      value: backend_kuma-demo_svc_3001
+                  origin:
+                    - mesh: default
+                      name: on-service
+                      type: MeshTimeout
+        """
+
+      When I visit the "/meshes/default/data-planes/dataplane-1/policies" URL
+
+      Then the "$policies-view" element contains "MeshHTTPRoute"
+      And the "$policies-view" element contains "MeshTimeout"
+
+      When I click the "$to-rule-item:nth-child(1) [data-testid='accordion-item-button']" element
+
+      Then the "$policies-view" element contains "kuma.io/service:other-svc"
+      And the "$policies-view" element contains "kuma.io/service:backend_kuma-demo_svc_3001"
+
+      When I click the "$to-rule-item:nth-child(2) [data-testid='accordion-item-button']" element
+
+      Then the "$policies-view" element contains "__rule-matches-hash__:waFsoIISmZDTfFWYqxvY265/GASHYEWvHQTwMh/bpuU= and kuma.io/service:backend_kuma-demo_svc_3001"
+      And the "$policies-view" element contains "!__rule-matches-hash__:waFsoIISmZDTfFWYqxvY265/GASHYEWvHQTwMh/bpuU= and kuma.io/service:backend_kuma-demo_svc_3001"
+
+    Scenario: Policies tab has expected content (MeshTimeout with from & to rules)
+      Given the environment
+        """
+        KUMA_DATAPLANEPROXYRULE_ENABLED: false
+        KUMA_DATAPLANERULE_COUNT: 1
+        KUMA_DATAPLANETORULE_COUNT: 3
+        KUMA_DATAPLANEFROMRULE_COUNT: 1
```

Comment:
Ouch these names read really awkwardly, umm 🤔 , I have some questions as I'm not that familiar with the data here:

1. Do we ever test something where we absolutely need the TO and FROM rules to be different?
2. Wondering if we can just use `KUMA_RULE`, I know we have different sort of rules, but I don't think we'll ever need to worry about trying to set the amount of RULES for dataplanes and accidentally setting it for roles also? Definitely still wondering about this one, I'm not sure what I think either.
3. If we want differently named _RULES (see above) are these POLICY_RULES? Maybe not.

I think all in all we've unfortunately not used `_` in other places, but I think the original thinking was to not use `_` for model/entity names but to use `_` for PARENT_CHILD things which would make this read better overall (so `KUMA_FAULTINJECTIONS` but `KUMA_DATAPLANE_RULES`)

Anyway as you can see from this comment I'm not entirely sure myself here, and its just a naming thing, but still, these are really hard to look at. Let me know what you think on any of the above before we decide to do anything about it, if anything

(If anything above isn't clear please ask)

Reply:
> 1. Do we ever test something where we absolutely need the TO and FROM rules to be different?

Very much so. The meaning changes if for what in the test scenario is expected to be 2 to rules and no proxy and from rules to suddenly have from rules, for example. So we do need to control that on the level of "rule types", not just a "KUMA_RULE".

> 2\. Wondering if we can just use `KUMA_RULE`, I know we have different sort of rules, but I don't think we'll ever need to worry about trying to set the amount of RULES for dataplanes and accidentally setting it for roles also? Definitely still wondering about this one, I'm not sure what I think either.

We could consider removing the `DATAPLANE` prefix, but there is in fact the potential for clashes as rules also exist in an RBAC context, so I don’t think that’s worthwhile.

> 3\. If we want differently named _RULES (see above) are these POLICY_RULES? Maybe not.

Not sure I understand the question completely. The association of those env entries (and the rest of the code) with dataplanes is partially arbitrary: it could also belong to policies. But really, what’s more accurate is to say these entities exist on the relationship between dataplanes and policies. I picked dataplanes because that’s the scope in which the _rules API is defined.

> I think all in all we've unfortunately not used `_` in other places, but I think the original thinking was to not use `_` for model/entity names but to use `_` for PARENT_CHILD things which would make this read better overall (so `KUMA_FAULTINJECTIONS` but `KUMA_DATAPLANE_RULES`)

I just stuck to the pattern that the env entries started out with which was `KUMA_` followed by more or less resource/entity name and finally a modifier for its "nature" or "type": `_COUNT` or `_ENABLED`.

We could just add underscores to separate all words. Then at least this becomes more readable.

I’m not set on keeping the `DATAPLANE` bit in there, but I do kinda want to avoid clashes with other concepts like RBAC.

Reply:
For the time being, I changed the names to always separate words by underscore.